packages/pages/src/vite-plugin/modules/plugin.ts

```diff
@@ -117,9 +123,44 @@ export const buildModules = async (
         }),
       ],
     });
+    editAssetsPaths(
+      `${rootFolders.dist}/${subfolders.modules}/${moduleName}.umd.js`,
+      getModuleDomain(fileInfo.path),
+      subfolders
+    );
   }
 };
 
+/**
+ * Replaces asset paths with absolute paths based on hosted domain.
+ * Ex. if the asset path is "src/assets/foo.svg", it'll be changed to
+ * "<hostedDomain>/modules/assets/foo.svg".
+ *
+ * @param umdPath dist/modules/<moduleName>.umd.js
+ * @param domain if given by user
+ * @param subfolders of the ProjectStructure
+ */
+const editAssetsPaths = (
```

Comment:
Let's try Vite's [renderBuiltUrl](https://vitejs.dev/guide/build.html#advanced-base-options). I believe this is exactly what we want instead of building and parsing/rewriting the final output.